js/data/bucket/symbol_bucket.js

```diff
@@ -118,11 +118,12 @@ class SymbolBucket {
     }
 
     populate(features, options) {
-        const layout = this.layers[0].layout;
+        const layer = this.layers[0];
+        const layout = layer.layout;
         const textFont = layout['text-font'];
         const iconImage = layout['icon-image'];
 
-        const hasText = layout['text-field'] && textFont;
+        const hasText = textFont && (!layer.isLayoutValueFeatureConstant('text-field') || layout['text-field']);
```

Comment:
> If the text-field property is data-driven, the value of layout['text-field'] is not defined.

@jfirebaugh ah, whoops - I made the bad assumption that `layout` contained the style JSON.